zipline/gens/tradesimulation.py

```diff
@@ -186,51 +190,135 @@ def once_a_day(midnight_dt):
+        as of the look-back date (the last day of the simulation).  Prices are
+        fully adjusted for dividends, splits, and mergers.
+
+        Returns
+        -------
+        A pd.Series, indexed by trading day, whose values represent the %
+        change from close to close.
         """
-        rvars = self.algo.recorded_vars
-        if self.algo.perf_tracker.emission_rate == 'daily':
-            perf_message = \
-                self.algo.perf_tracker.handle_market_close_daily()
-            perf_message['daily_perf']['recorded_vars'] = rvars
-            yield perf_message
-
-        elif self.algo.perf_tracker.emission_rate == 'minute':
-            # close the minute in the tracker, and collect the daily message if
-            # the minute is the close of the trading day
-            minute_message, daily_message = \
-                self.algo.perf_tracker.handle_minute_close(dt)
-
-            # collect and yield the minute's perf message
-            minute_message['minute_perf']['recorded_vars'] = rvars
-            yield minute_message
-
-            # if there was a daily perf message, collect and yield it
-            if daily_message:
-                daily_message['daily_perf']['recorded_vars'] = rvars
-                yield daily_message
+        if sid is None:
+            # get benchmark info from trading environment
+            benchmark_series = \
+                env.benchmark_returns[trading_days[0]:trading_days[-1]]
+        else:
+            # check if this security has a stock dividend.  if so, raise an
+            # error suggesting that the user pick a different asset to use
+            # as benchmark.
+            stock_dividends = \
+                data_portal.get_stock_dividends(sid, trading_days)
+
+            if len(stock_dividends) > 0:
+                raise InvalidBenchmarkAsset(
+                    sid=str(sid),
+                    dt=stock_dividends[0]["ex_date"]
+                )
+
+            benchmark_asset = env.asset_finder.retrieve_asset(sid)
+            if benchmark_asset.start_date > trading_days[0]:
+                # the asset started trading after the first simulation day
+                raise BenchmarkAssetNotAvailableTooEarly(
+                    sid=str(sid),
+                    dt=trading_days[0],
+                    start_dt=benchmark_asset.start_date
+                )
+
+            if benchmark_asset.end_date < trading_days[-1]:
+                # the asset stopped trading before the last simulation day
+                raise BenchmarkAssetNotAvailableTooLate(
+                    sid=str(sid),
+                    dt=trading_days[0],
+                    end_dt=benchmark_asset.end_date
+                )
+
+            # get the window of close prices for benchmark_sid from the last
+            # trading day of the simulation, going up to one day before the
+            # simulation start day.
+            benchmark_series = data_portal.get_history_window(
```

Comment:
It is good that we can use history here.
However, the usage of this function would read clearer (at least to me) in this context if the parameters were start_dt and end_dt, instead of `len(trading_days) + 1`.

(I could concede that that may be just shuffling around the abstraction.)